actnow/profiles/permissions.py

```diff
@@ -0,0 +1,15 @@
+from rest_framework.permissions import SAFE_METHODS, BasePermission
+
+
+class IsOwnerOrReadOnly(BasePermission):
+    def has_object_permission(self, request, view, obj):
+        if request.method in SAFE_METHODS:
+            return True
+        return bool(obj.user.id == request.user.id)
+
+
+class IsUserProfile(BasePermission):
+    def has_permission(self, request, view):
+        user_id = request.data.get("user", -1)
+        # Prevent a user from creating other users profiles.
+        return int(user_id) == request.user.id
```

Comment:
Shouldn't this be enforced in the view or serializer i.e. when creating user profile, user will always be set to the request.user?

Reply:
yes, this is enforced in the view(when we determine which permission to be used for which action). Or do you mean it should be enforced inside the `get_querysert`?

Reply:
I mean rather than this being a permission, wouldn't it make more sense for it to never be allowed to happen?

Reply:
Ahh. I see your point.